app/src/main/java/eu/kanade/tachiyomi/source/SourceManager.kt

```diff
@@ -23,20 +30,53 @@ open class SourceManager(private val context: Context) {
 
     fun getOrStub(sourceKey: Long): Source {
         return sourcesMap[sourceKey] ?: stubSourcesMap.getOrPut(sourceKey) {
-            StubSource(sourceKey)
+            val sourceData: SourceData =
+                db.getSourceData(sourceKey).executeAsBlocking() ?:
+                extensionManager.getSourceData(sourceKey).also { registerStubSource(it!!, true) } ?:
+                SourceData.create()
+                    .apply { id = sourceKey }
+            StubSource(sourceData)
```

Comment:
![image](https://user-images.githubusercontent.com/6576096/148538672-7143be0c-e546-4e36-b19a-8333dd35e855.png)

Perhaps use `?.let` and return early instead of using the elvis operator along with `also` and `apply`

```suggestion
            db.getSourceData(sourceKey).executeAsBlocking()?.let {
                return@getOrPut StubSource(it)
            }
            extensionManager.getSourceData(sourceKey)?.let {
                registerStubSource(it, true)
                return@getOrPut StubSource(it)
            }
            StubSource(SourceData.create().apply { id = sourceKey })
```

---

```kt
.also { registerStubSource(it!!, true) }
```

Also that `!!` operator will throw an `NullPointerException` when you have a null value, you want to do `?.also`  instead of just `.also`

Reply:
agree readability over trying to string together as many kotlin keywords as possible